toxcore/group.c

```diff
@@ -29,24 +29,80 @@
 
 #include "util.h"
 
-/* return 1 if the groupnumber is not valid.
- * return 0 if the groupnumber is valid.
- */
-static uint8_t groupnumber_not_valid(const Group_Chats *g_c, int groupnumber)
+typedef ptrdiff_t aint;
+
+#define CLOSEST( i ) ((g->closest_peers_entry & (1ull << i)) != 0)
+#define CLOSESTSET( i ) { g->closest_peers_entry |= (1ull << i); }
+
+typedef enum {
+    UNS_NONE,
+    UNS_TEMP,
+    UNS_FOREVER,
+
+} Unsubscribe;
+
+/* Interval in seconds to send ping messages */
+#define GROUP_PING_INTERVAL 20
+
+#define GROUP_MESSAGE_PING_ID 0
+#define GROUP_MESSAGE_UNSUBSCRIBE_ID 1
+#define GROUP_MESSAGE_NICKNAME_ID 2
+#define GROUP_MESSAGE_NEW_PEER_ID 16
+#define GROUP_MESSAGE_KILL_PEER_ID 17
+#define GROUP_MESSAGE_NAME_ID 48
+#define GROUP_MESSAGE_TITLE_ID 49
```

Comment:
Would it make sense for these to be enums as well? Also `INVITE_*` and `PEER_*`.